libs/log_parser.rb

```diff
@@ -13,5 +17,34 @@ def read_first_line
       raise "input file not found or does not exists"
     end
   end
+
+  def count_lines
+    @lines = File.foreach(@file_name).count
+  end
+
+  def parse_file
+    count_lines
+    player_filter
+    obj = {
+      "#{@file_name.split("/").last}": {
+        lines: @lines,
+        players: @players
+      }
+    }
+    obj_json = JSON.pretty_generate(obj)
+  end
+
+  def player_filter
+    File.readlines(@file_name).each do |line|
+      if line.include? "killed"
+        player = line.split("killed ").last.split(" by").first
```

Comment:
seria bom extrair essa lógica em uma função para deixar o seu código mais declarativo
o metodo get_kill_players deve ser um metodo privado
```suggestion
        player = get_kill_players(line)
```
```
        def get_kill_players(line)
          line.split("killed ").last.split(" by").first
        end
```